### PR TITLE
[Core][Geometry] Projection and closest point: correct API and add new methods

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**Description**
This correct the API of the new closest point method according to the discussion in #8491. I also took the chance to add the interface to the new ones, which throw an error as discussed.

In a further PR we can deprecate the old ones.

@tteschemacher can you please check that the changes I've done in your geometries are correct?

**Changelog**
New projection and closest point methods to the interface.

### CODE REVIEW

File: kratos/utilities/nurbs_utilities/projection_nurbs_geometry_utilities.h [hunk 1]
```diff
@@ -95,9 +95,9 @@ class ProjectionNurbsGeometryUtilities
             if (norm_2(delta_t * derivatives[1]) < Accuracy)
                 return true;
 
-            // Check if the parameter gets out of its interval of definition and if so clamp it 
+            // Check if the parameter gets out of its interval of definition and if so clamp it
             // back to the boundaries
-            int check = rGeometry.ClosestPointLocalToLocalSpace(rParameterLocalCoordinates);
+            int check = rGeometry.ClosestPointLocalToLocalSpace(rParameterLocalCoordinates, rParameterLocalCoordinates);
```
Comment (tteschemacher) [inline kratos/utilities/nurbs_utilities/projection_nurbs_geometry_utilities.h:100]: are you sure that this is not a bit evil?

**Comment by tteschemacher** [comment]
Hey @rubenzorrilla ,

thank you very much for this implementation. It looks great to me.

File: kratos/geometries/geometry.h [hunk 2]
```diff
@@ -2560,6 +2560,60 @@ class Geometry
...
+     * It shows only if the breaking criteria, defined by the tolerance is reached.
+     * This function requires an initial guess, provided by rProjectionPointLocalCoordinates.
+     * This function can be a very costly operation.
+     * @param rPointLocalCoordinates Global coordinates of the point to be projected
+     * @param rProjectionPointLocalCoordinates Projection point local coordinates. This should be initialized with the initial guess
+     * @param Tolerance Accepted orthogonal error
+     * @return int 0 -> failed
+     *             1 -> converged
+     */
+    virtual int ProjectionPoinGlobalToLocalSpace(
```
Comment (tteschemacher) [inline kratos/geometries/geometry.h]:
```suggestion
    virtual int ProjectionPointGlobalToLocalSpace(
```

little typo

File: kratos/utilities/nurbs_utilities/projection_nurbs_geometry_utilities.h (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/utilities/nurbs_utilities/projection_nurbs_geometry_utilities.h:100]: I think there shouldn't be any problem (note that the first reference is taken as `const`). I think that doing a copy with the unique purpose of passing it to the `ClosestPointLocalToLocalSpace` wasn't very good idea...

**Review by tteschemacher — APPROVED** [review]
To me it looks good. Once again, thanks!

**Pull request merged**